File: sentinel-frontend/src/types/alertrule.ts
export type AlertOperator = ">" | "<" | ">=" | "<=";
export type AlertSeverity = "LOW" | "MEDIUM" | "CRITICAL";
export type MetricName = "cpu" | "ram" | "disk";

export interface AlertRule {
    rule_id: string;
    tenant_id: string; // tenant identifier
    host_id: string;   // host identifier
    metric_name: MetricName;
    threshold: number;
    operator: AlertOperator;
    severity: AlertSeverity;
    enabled: number; // 1 | 0
}

File: sentinel-alert/src/main/java/org/example/sentinelalert/Model/AlertRule.java
package org.example.sentinelalert.Model;

import lombok.Data;

@Data
public class AlertRule {
    private String ruleId;
    private String metricName;
    private float threshold;
    private String operator;
    private String severity;
    private boolean enabled;
    private String tenantId;
    private String hostId;
}
